destination_test.go

```diff
@@ -1,15 +1,378 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package vectorstore
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"os"
 	"testing"
 
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
 	"github.com/matryer/is"
+	"github.com/rs/zerolog"
+	"github.com/sashabaranov/go-openai"
 )
 
+var getOpenAIApiKey = envGetter("OPENAI_API_KEY")
+
 func TestTeardown_NoOpen(t *testing.T) {
 	is := is.New(t)
 	con := NewDestination()
 	err := con.Teardown(context.Background())
 	is.NoErr(err)
 }
+
+func TestWrite_Create(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file := newTestFile()
+	rec := file.recCreate()
+
+	written, err := dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	assertFileWrittenAndUnique(ctx, t, file, vectorStoreID)
+}
+
+func TestWrite_MultipleWritesWithSameKeyDontDuplicate(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file1, file2 := newTestFile(), newTestFile()
```

Comment:
didn't know you could do this

Reply:
And they say go isn't expressive :man_shrugging: